### PR TITLE
contract: Reimplement legacy administrative contract validation

### PR DESCRIPTION
**Pull request by cyrossignol**
This adds a minimal legacy contract validation adaptation that verifies the signature of version 1 administrative contracts to replace the bulky contract signature parsing and verification code removed in #1933. It prevents invalid historical testnet contracts from interfering with the initial sync. I will sleep better with this in place for mainnet as well.

### CODE REVIEW

**Review by jamescowens — APPROVED** [review]
utACK.

File: src/gridcoin/contract/contract.cpp [hunk 1]
```diff
@@ -330,6 +330,58 @@ class Dispatcher
...
+    const ContractPayload payload = contract.m_body.AssumeLegacy();
+    const auto& body = static_cast<const LegacyPayload&>(*payload);
+
+    CHashWriter hasher(SER_NETWORK, PROTOCOL_VERSION);
+    hasher << type_string << body.m_key << body.m_value;
+
+    CKey key;
+    key.SetPubKey(CWallet::MasterPublicKey());
+
+    return key.Verify(hasher.GetHash(), decoded);
```
Comment (div72) [inline src/gridcoin/contract/contract.cpp]:
```suggestion
    return key.Verify(hasher.GetHash(), sig);
```

**Comment by jamescowens** [comment]
@cyrossignol is doing another sync from zero test on this on mainnet and testnet before we merge this.

**Comment by jamescowens** [comment]
@cyrossignol's sync from zero for both mainnet and testnet was successful. Merging.

**Pull request merged**